protocol/contracts/v2/TreasuryPriceIndexOracle.sol

```diff
@@ -21,98 +20,125 @@ contract TreasuryPriceIndexOracle is ITreasuryPriceIndexOracle, TempleElevatedAc
+        }
+     }
 
     /**
-     * @notice Set the number of seconds to elapse before a new TPI will take effect.
+     * @notice Set the maximum allowed TPI change on any single `setTreasuryPriceIndexAt()`, in absolute terms
+     * between the TPI as of now and the targetTpi
+     * @dev 18 decimal places, 0.20e18 == $0.20
      */
-    function setTpiCooldown(uint32 cooldownSecs) external override onlyElevatedAccess {
-        emit TpiCooldownSet(cooldownSecs);
-        tpiData.cooldownSecs = cooldownSecs;
+    function setMaxTreasuryPriceIndexDelta(uint96 maxDelta) external override onlyElevatedAccess {
+        emit MaxTreasuryPriceIndexDeltaSet(maxDelta);
+        maxTreasuryPriceIndexDelta = maxDelta;
     }
 
     /**
-     * @notice Set the maximum allowed TPI change on any single `setTreasuryPriceIndex()`, in absolute terms.
-     * @dev 18 decimal places, 0.20e18 == $0.20
+     * @notice Set the minimum time delta required for TPI to reach it's target value when 
+     * `setTreasuryPriceIndexAt()` is called.
+     * @dev In seconds.
      */
-    function setMaxTreasuryPriceIndexDelta(uint256 maxDelta) external override onlyElevatedAccess {
-        emit MaxTreasuryPriceIndexDeltaSet(maxDelta);
-        maxTreasuryPriceIndexDelta = maxDelta;
+    function setMinTreasuryPriceIndexTargetTimeDelta(uint32 minTargetTimeDelta) external override onlyElevatedAccess {
+        emit MinTreasuryPriceIndexTargetTimeDeltaSet(minTargetTimeDelta);
+        minTreasuryPriceIndexTargetTimeDelta = minTargetTimeDelta;
     }
 
     /**
-     * @notice Set the Treasury Price Index (TPI)
-     * @dev 18 decimal places, 1.05e18 == $1.05
+     * @notice Set the target TPI which will incrementally increase from it's current value to `targetTpi`
+     * between now and `targetTime`.
+     * @dev targetTime is unixtime, targetTpi is 18 decimal places, 1.05e18 == $1.05
      */
-    function setTreasuryPriceIndex(uint96 value) external override onlyElevatedAccess {
-        // If the cooldownSecs hasn't yet passed since the last update, then this will still
-        // refer to the `previousTpi` value
-        uint96 _oldTpi = treasuryPriceIndex();
-        uint96 _newTpi = value;
-
-        unchecked {
-            uint256 _delta = (_newTpi > _oldTpi) ? _newTpi - _oldTpi : _oldTpi - _newTpi;
-            if (_delta > maxTreasuryPriceIndexDelta) revert BreachedMaxTpiDelta(_oldTpi, _newTpi, maxTreasuryPriceIndexDelta);
-        }
+    function setTreasuryPriceIndexAt(uint96 targetTpi, uint32 targetTime) external override onlyElevatedAccess {
+        uint96 _currentTpi = treasuryPriceIndex();
+        uint32 _now = uint32(block.timestamp);
+        int96 _tpiDelta = int96(targetTpi) - int96(_currentTpi);
+
+        // targetTime must be at or after (now + minTreasuryPriceIndexTargetTimeDelta)
+        if (targetTime < _now + minTreasuryPriceIndexTargetTimeDelta) revert BreachedMinDateDelta(targetTime, _now, minTreasuryPriceIndexTargetTimeDelta);
```

Comment:
I think the `max-delta` and the `min-duration` should not be two variables to validate independently, but instead, the ratio between them is what should be validated in the requirements. The value of one without the other is meaningless, so I think it is not fully correct that their caps have individual setters (`setMaxTreasuryPriceIndexDelta()` and `setMinTreasuryPriceIndexTargetTimeDelta()`) without any constraint between them when you set them. 

Put in other words:
- If the max delta tpi is 0.2$, it means something completely different if the minTimeDelta is 1 day or 1 year.
- Similarly, if the minTimeDelta is 1 month, it means something completely different if the max delta tpi is 0.2$ or 2$

Therefore, what I would suggest is that the slope of dripping (the ratio between the tpiDelta and timeDelta) is what is validated in the requirements. 

Therefore, instead of validating the inputs against `maxTreasuryPriceIndexDelta` and `minTreasuryPriceIndexTargetTimeDelta` individually, I would have something like `max-absolute-tpi-slope`, expressed for instance like `maxTpiDeltaOverOneYear` (in absolute, as it can be negative).

Then, when you call `setTreasuryPriceIndexAt()`, your validation is not done against the absolute values of `targetTpiDelta` and `targetTimeDelta` independently, but on their RELATIONSHIP (on the slope: `targetTpiDelta` / `targetTimeDelta`). 

Example calculation below:

```solidity
	// absTpiDeltaOverOneYear = (365 days) * absTargetTpiDelta / targetTimeDelta;  // TPI slope over one year

	// the overall condition to be fulfilled in the tpi setter:
	//     require(absTpiDeltaOverOneYear < maxTpiDeltaOverOneYear);
	//     require((365 days) * absTargetTpiDelta / targetTimeDelta < maxTpiDeltaOverOneYear);
	// which can be rewritten as the following (to avoid division rounding headaches):
	require((365 days) * targetTpi < targetTimeDelta * maxTpiDeltaOverOneYear);
```

Reply:
Another alternative to enforce this relationship boundaries would be to keep the two individual caps `maxTreasuryPriceIndexDelta` and `minTreasuryPriceIndexTargetTimeDelta`, and do the slope-requirement validation in both setters: 
- `setMaxTreasuryPriceIndexDelta()`
- `setMinTreasuryPriceIndexTargetTimeDelta()` 

So you know anytime you change any of the two, your know your slope is not going nuts

Reply:
And of course, you can still keep the individual constrains like for min-duration if you'd like.

Reply:
Good idea, I've updated here
https://github.com/TempleDAO/temple/pull/1096/commits/27c667373ccc4302b0fe3a9f1d25acc4c5e018c3

we'd have something like:
* min 2 week duration
* max $1 TPI delta
* max slope of 1c/day

Reply:
looks great